02.calendar/cal.js

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env node
+import minimist from "minimist";
+
+const HEADER_WIDTH = 14;
+
+const run = () => {
+  const today = new Date();
+  const currentYear = today.getFullYear();
+  const currentMonth = today.getMonth() + 1;
+
+  const parseOptions = minimist(process.argv.slice(2), {
+    string: ["-y", "-m"],
+    alias: {
+      y: "year",
+      m: "month",
+    },
+    default: {
+      y: currentYear,
+      m: currentMonth,
+    },
+  });
+
+  const year = parseOptions.year;
+  const month = parseOptions.month;
+
+  return [
+    generateCalendarHeader(year, month),
+    generateWeekHeader(),
+    ...generateMonthDates(year, month),
+  ].join("\n");
+};
+
+const generateCalendarHeader = (year, month) => {
+  const monthName = getFirstDateOfMonth(year, month).toLocaleString("en-US", {
+    month: "short",
+  });
+  return `${monthName} ${year}`.padStart(HEADER_WIDTH, " ");
+};
+
+const getFirstDateOfMonth = (year, month) => {
+  return new Date(year, month - 1, 1);
+};
+
+const generateWeekHeader = () => {
+  return "Su Mo Tu We Th Fr Sa";
+};
+
+const generateMonthDates = (year, month) => {
+  const dayInMonth = getLastDateOfMonth(year, month).getDate();
```

Comment:
```suggestion
  const lastDay = getLastDayOfMonth(year, month);
```

みたいな感じにした方がシンプルでいいかもしれません。

Reply:
そちらの変数名の方がわかりやすいです✨
修正します〜